### PR TITLE
Added Storage Account Container deployment resource and added the deploymentUrlBase variable

### PR DESCRIPTION
**Pull request by dalemorson**
- Added the `deploymentUrlBase` to variable section
- Updated uri links in each linked deployment resource to use the `deploymentUrlBase` variable
- Added a Storage Account Container deployment resource in preparation for removing the PowerShell `Create Template Container` task in the das-assessor-service release pipeline as it is no longer required and will enable the removal of the ESFA.InfrastructureAutomation package.

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -442,14 +453,42 @@
                 }
             }
         },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "[concat('StorageContainer', parameters('containerArray')[copyIndex()].name)]",
```
Comment (jack-education) [inline azure/template.json]:
```suggestion
            "name": "[concat('storage-container', parameters('containerArray')[copyIndex()].name)]",
```

File: azure/template.json [hunk 2]
```diff
@@ -442,14 +453,42 @@
...
+            "type": "Microsoft.Resources/deployments",
+            "condition": "[greater(length(parameters('containerArray')), 0)]",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'), 'storage-container.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
```
Comment (jack-education) [inline azure/template.json]: Maybe good to hardcode the publicAccess parameter as None so it is clear

File: azure/template.json [hunk 3]
```diff
@@ -442,14 +453,42 @@
...
+                    "storageAccountName": {
+                        "value": "[parameters('storageAccountName')]"
+                    },
+                    "containerName": {
+                        "value": "[parameters('containerArray')[copyIndex()].name]"
+                    }
+                }
+            },
+            "copy": {
+                "name": "containercopy",
```
Comment (jack-education) [inline azure/template.json]:
```suggestion
                "name": "storagecontainercopy",
```

File: azure/template.json (hunk 2 above)
Comment (dalemorson) [reply azure/template.json]: Added new commit hard coding it to none.

**Pull request merged**